sdk/appconfiguration/Azure.Data.AppConfiguration/tests/samples/Sample11_SettingsSnapshot.cs

```diff
@@ -100,27 +183,29 @@ public void GetSnapshots()
 
-            List<ConfigurationSettingFilter> firstSnapshotFilter = new(new ConfigurationSettingFilter[] { new ConfigurationSettingFilter(firstSetting.Key) });
-            ConfigurationSettingsSnapshot createdfirstSnapshot = client.CreateSnapshot("first_snapshot", new ConfigurationSettingsSnapshot(firstSnapshotFilter));
-            Console.WriteLine($"Created configuration setting snapshot is: {createdfirstSnapshot}");
+            List<SnapshotSettingFilter> firstSnapshotFilter = new(new SnapshotSettingFilter[] { new SnapshotSettingFilter(firstSetting.Key) });
+            CreateSnapshotOperation createfirstSnapshotOperation = client.CreateSnapshot(WaitUntil.Completed, "first_snapshot", new ConfigurationSettingsSnapshot(firstSnapshotFilter));
+            ConfigurationSettingsSnapshot createdFirstSnapshot = createfirstSnapshotOperation.Value;
+            Console.WriteLine($"Created configuration setting snapshot is: {createdFirstSnapshot}");
 
-            Assert.NotNull(createdfirstSnapshot);
-            Assert.AreEqual("first_snapshot", createdfirstSnapshot.Name);
+            Assert.NotNull(createdFirstSnapshot);
+            Assert.AreEqual("first_snapshot", createdFirstSnapshot.Name);
 
-            List<ConfigurationSettingFilter> secondSnapshotFilter = new(new ConfigurationSettingFilter[] { new ConfigurationSettingFilter(secondSetting.Key) });
-            ConfigurationSettingsSnapshot createdsecondSnapshot = client.CreateSnapshot("second_snapshot", new ConfigurationSettingsSnapshot(secondSnapshotFilter));
-            Console.WriteLine($"Created configuration setting snapshot is: {createdsecondSnapshot}");
+            List<SnapshotSettingFilter> secondSnapshotFilter = new(new SnapshotSettingFilter[] { new SnapshotSettingFilter(secondSetting.Key) });
+            CreateSnapshotOperation createdSecondSnapshotOperation = client.CreateSnapshot(WaitUntil.Completed, "second_snapshot", new ConfigurationSettingsSnapshot(secondSnapshotFilter));
+            ConfigurationSettingsSnapshot createdSecondSnapshot = createdSecondSnapshotOperation.Value;
+            Console.WriteLine($"Created configuration setting snapshot is: {createdSecondSnapshot}");
 
-            Assert.NotNull(createdsecondSnapshot);
-            Assert.AreEqual("second_snapshot", createdsecondSnapshot.Name);
+            Assert.NotNull(createdSecondSnapshot);
+            Assert.AreEqual("second_snapshot", createdSecondSnapshot.Name);
 
             // #region Snippet:Sample_GetSnapshots
             var count = 0;
-            foreach (ConfigurationSettingsSnapshot item in client.GetSnapshots())
-            {
-                count++;
-                Console.WriteLine($"Name {item.Name} status {item.Status}");
-            }
+            //foreach (ConfigurationSettingsSnapshot item in client.GetSnapshots())
+            //{
+            //    count++;
+            //    Console.WriteLine($"Name {item.Name} status {item.Status}");
+            //}
```

Comment:
Why did we comment out this loop?

Reply:
Since this is in the project, the compiler could no longer distinguish between the public GetSnapshots(string,string,IEnumerable<SnapshotFields>,IEnumerable<SnapshotStatus, CancellationToken) and the public GetSnapshots(string, string, IEnumerable<string>,IEnumerable<string>,RequestContext)

Reply:
We may need to update this to pass in a specific parameter to help distinguish between them. FWIW this isn't a problem for the public API

Reply:
See response [here](https://github.com/Azure/azure-sdk-for-net/pull/34583#discussion_r1156923716).

I'll uncomment this code and fix the method ambiguity issue